src/bun.js/bindings/OnigurumaRegExp.cpp

```diff
@@ -46,6 +54,39 @@ static  WTF::String to16Bit(JSValue jsValue, JSC::JSGlobalObject *globalObject,
     return to16Bit(jsString, globalObject);
 }
 
+static WTF::String extendMultibyteHexCharacters(const WTF::String &string) {
+    WTF::StringBuilder sb;
+    for (int i = 0; i < string.length(); i++) {
+        if (string.characterAt(i) == '\\') {
+            while (string.characterAt(i) == '\\') {
+                if (i + 1 < string.length() && string.characterAt(i + 1) == 'x') {
```

Comment:
would be slightly better to put `string.length()` in a variable up top since that won't change here and it's likely a pointer each time